File: pkg/model/rating.go
package model

import (
	"errors"

	"github.com/uptrace/bun"
)

type Rating struct {
	bun.BaseModel
	ID      int64  `json:"id" bun:",pk,autoincrement"`
	Stars   int    `json:"stars"`
	UserID  int64  `json:"-"`
	User    *User  `json:"-" bun:"rel:belongs-to,join:user_id=id"`
	PizzaID int64  `json:"pizza_id"`
	Pizza   *Pizza `json:"-" bun:"rel:belongs-to,join:pizza_id=id"`
}

func (r *Rating) Validate() error {
	if r.Stars < 1 || r.Stars > 5 {
		return errors.New("number of stars must be between 1 and 5 (inclusive)")
	}

	return nil
}

File: Mobiles/flutter/lib/models/rating.dart
class Rating {
  final int id;
  final int pizzaId;
  final int stars;
  final int userId;

  Rating({
    required this.id,
    required this.pizzaId,
    required this.stars,
    required this.userId,
  });

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      id: json['id'] as int,
      pizzaId: json['pizza_id'] as int,
      stars: json['stars'] as int,
      userId: json['user_id'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'pizza_id': pizzaId,
      'stars': stars,
    };
  }
}

